File: internal/indexer/testdata/go/sample.go
package sample

import "fmt"

// MaxRetries is a package-level constant.
const MaxRetries = 3

// DefaultName is a package-level variable.
var DefaultName = "world"

// Greeter is an interface for greeting.
type Greeter interface {
	Greet(name string) string
}

// Person represents a person with a name and age.
type Person struct {
	Name string
	Age  int
}

// NewPerson creates a new Person.
func NewPerson(name string, age int) *Person {
	return &Person{Name: name, Age: age}
}

// Greet implements the Greeter interface.
func (p *Person) Greet(name string) string {
	return fmt.Sprintf("Hello, %s! I'm %s.", name, p.Name)
}

// SayHello is a standalone function.
func SayHello() {
	p := NewPerson(DefaultName, 30)
	msg := p.Greet("friend")
	fmt.Println(msg)
	for i := 0; i < MaxRetries; i++ {
		fmt.Println(p.Greet(DefaultName))
	}
}

File: internal/indexer/testdata/python/sample.py
from typing import List, Optional

MAX_RETRIES = 3
default_name = "world"

class Greeter:
    def greet(self, name: str) -> str:
        raise NotImplementedError

class Person(Greeter):
    def __init__(self, name: str, age: int):
        self.name = name
        self.age = age

    def greet(self, name: str) -> str:
        return f"Hello, {name}! I'm {self.name}."

    @property
    def display_name(self) -> str:
        return self.name.upper()

def say_hello():
    p = Person(default_name, 30)
    msg = p.greet("friend")
    print(msg)
    for i in range(MAX_RETRIES):
        print(p.greet(default_name))
    print(p.display_name)

def format_name(name: str) -> str:
    return name.strip().title()

if __name__ == "__main__":
    say_hello()
    result = format_name(default_name)
    print(result)
